pallets/runtime/tests/src/bridge.rs

```diff
@@ -12,790 +9,515 @@ use frame_support::{
+    });
+}
+
+#[test]
+fn freeze_txs() {
+    test_with_controller(&do_freeze_txs);
+}
+
+fn do_freeze_txs(signers: &[Public]) {
+    let no_admin = Origin::signed(*signers.iter().next().unwrap());
+
+    // Create some txs and register the recipients' balance.
+    let txs = proposals(AMOUNT)
+        .into_iter()
+        .map(|(_, p)| signers_approve_proposal(*p, signers))
+        .collect::<Vec<_>>();
+    let init_balances = txs
+        .iter()
+        .map(|tx| Balances::total_balance(&tx.recipient))
+        .collect::<Vec<_>>();
+
+    advance_block(Bridge::timelock());
+
+    // Freeze all txs except the first one.
+    let frozen_txs = txs.iter().skip(1).cloned().collect::<Vec<_>>();
+    assert!(!frozen_txs.is_empty());
+    assert_noop!(
+        Bridge::freeze_txs(no_admin.clone(), frozen_txs.clone()),
+        Error::BadAdmin
     );
+    assert_ok!(Bridge::freeze_txs(signed_admin(), frozen_txs.clone()));
     next_block();
-    assert_eq!(alices_balance(), starting_alices_balance);
-    next_block();
-    assert_eq!(alices_balance(), starting_alices_balance);
-    next_block();
-    assert_eq!(alices_balance(), starting_alices_balance);
-    next_block();
-    assert_eq!(System::block_number(), unlock_block_number);
-    // Still no issue, rate limit reached
-    assert_eq!(alices_balance(), starting_alices_balance);
-    assert_ok!(Bridge::force_handle_bridge_tx(
-        admin.clone(),
-        bridge_tx.clone()
-    ));
-    // Mint successful after force handle
-    assert_eq!(alices_balance(), starting_alices_balance + amount);
-    assert_eq!(
-        Bridge::bridge_tx_details(AccountKeyring::Alice.public(), &1).execution_block,
-        unlock_block_number
-    );
-    assert_eq!(
-        Bridge::bridge_tx_details(AccountKeyring::Alice.public(), &1).status,
-        BridgeTxStatus::Handled
-    );
+
+    // Double check that first TX is done, and any other is frozen.
+    let tx = txs.iter().next().unwrap();
+    ensure_tx_status(tx.recipient, tx.nonce, BridgeTxStatus::Handled);
+
+    frozen_txs.iter().for_each(|tx| {
+        let _ = ensure_tx_status(tx.recipient, tx.nonce, BridgeTxStatus::Frozen);
+    });
+
+    // Unfreeze frozen TXs
+    assert_noop!(
+        Bridge::unfreeze_txs(no_admin, frozen_txs.clone()),
+        Error::BadAdmin
+    );
+    assert_ok!(Bridge::unfreeze_txs(signed_admin(), frozen_txs));
+
+    // Verify that all TXs are done and balances of owner are updated.
+    txs.into_iter()
+        .zip(init_balances.into_iter())
+        .for_each(|(tx, init_balance)| {
+            ensure_tx_status(tx.recipient, tx.nonce, BridgeTxStatus::Handled);
+            assert_eq!(
+                Balances::total_balance(&tx.recipient),
+                init_balance + AMOUNT
+            );
+        });
+}
+
+#[test]
+fn batch_propose_bridge_tx() {
+    test_with_controller(&do_batch_propose_bridge_tx);
+}
+
+fn do_batch_propose_bridge_tx(signers: &[Public]) {
```

Comment:
Still the `do_` prefix in this file. I'd just inline this into the function above.